File: src/main/java/com/rtomyj/skc/filter/RequestFilter.java
package com.rtomyj.skc.filter;

import com.rtomyj.skc.logging.Logging;
import org.jetbrains.annotations.NotNull;
import org.slf4j.MDC;
import org.springframework.stereotype.Component;
import org.springframework.web.filter.OncePerRequestFilter;

import javax.servlet.FilterChain;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;


@Component
public class RequestFilter extends OncePerRequestFilter {
    @Override
    protected void doFilterInternal(@NotNull final HttpServletRequest request, @NotNull final HttpServletResponse response, final FilterChain chain)
            throws ServletException, IOException
    {
        try
        {
            Logging.configureMDC(request);
            chain.doFilter(request, response);
        } finally
        {
            MDC.clear();
        }
    }
}

File: src/main/kotlin/com/rtomyj/skc/filter/RequestFilter.kt
package com.rtomyj.skc.filter

import com.rtomyj.skc.logging.Logging
import org.slf4j.MDC
import org.springframework.stereotype.Component
import org.springframework.web.filter.OncePerRequestFilter
import java.io.IOException
import javax.servlet.FilterChain
import javax.servlet.ServletException
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Component
class RequestFilter : OncePerRequestFilter() {
    @Throws(ServletException::class, IOException::class)
    override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, chain: FilterChain) {
        try {
            Logging.configureMDC(request)
            chain.doFilter(request, response)
        } finally {
            MDC.clear()
        }
    }
}